services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/MetersValidationSingleSwFlowSpec.groovy

```diff
@@ -0,0 +1,406 @@
+        assumeTrue("Unable to find required switches in topology", switches as boolean)
+
+        setup: "Select a #switchType switch and retrieve default meters"
+        def sw = switches.first()
+
+        when: "Create a flow"
+        def flow = flowHelper.addFlow(flowHelper.singleSwitchFlow(sw))
+        def createdCookies = сookiesWithMeter(sw.dpId)
+
+        and: "Delete created rules"
+        northbound.deleteSwitchRules(sw.dpId, DeleteRulesAction.IGNORE_DEFAULTS)
+
+        then: "Rule info is moved into the 'missing' section"
+        def switchValidateInfo = northbound.switchValidate(sw.dpId)
+        switchValidateInfo.rules.missing.containsAll(createdCookies)
+
+        and: "The rest fields in the 'rule' section are empty"
+        ruleSectionsAreEmpty(switchValidateInfo, ["proper", "excess"])
+
+        when: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        then: "Check that rules validate info is also deleted"
+        Wrappers.wait(WAIT_OFFSET) {
+            def switchValidateInfoAfterDelete = northbound.switchValidate(sw.dpId)
+            println(sw.dpId)
+            ruleSectionsAreEmpty(switchValidateInfoAfterDelete)
+            meterSectionsAreEmpty(switchValidateInfoAfterDelete)
+        }
+
+        where:
+        switchType   | switches
+        "Centec"     | getCentecSwitches()
+        "non-Centec" | getNonCentecSwitches()
+    }
+
+    @Unroll
+    def "Switch validation is able to detect rule info into the 'excess' section on a #switchType switch"() {
+        assumeTrue("Unable to find required switches in topology", switches as boolean)
+
+        setup: "Select a #switchType switch and retrieve default meters"
+        def sw = switches.first()
+
+        when: "Create a flow"
+        def flow = flowHelper.addFlow(flowHelper.singleSwitchFlow(sw))
+        def createdCookies = сookiesWithMeter(sw.dpId)
+
+        and: "Change cookies for the created flow directly in DB"
+        def newFlowCookies = [createdCookies[0] + 1, createdCookies[1] + 1]
+        /** at this point rules are set for given flow. Now update flow cookie directly via DB,
+         it is done just for moving rules from the 'proper' section into the 'excess' section*/
+        database.updateFlowCookie(flow.id, newFlowCookies[0], newFlowCookies[1])
+
+        then: "Rule validate info(original cookies) is moved into the 'excess' section"
+        def switchValidateInfo = northbound.switchValidate(sw.dpId)
+        switchValidateInfo.rules.excess.collect { it }.containsAll(createdCookies)
+
+        and: "Rule validate info(new cookies) is moved into the 'excess' section"
+        switchValidateInfo.rules.missing.collect { it }.containsAll(newFlowCookies)
+        ruleSectionsAreEmpty(switchValidateInfo, ["proper"])
+
+        when: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        then: "Check that rules validate info is also deleted"
```

Comment:
Don't understand how this test passes. Who removes excess rules? If this works, then sth is wrong in the way we update the db and the test is invalid. Flow deletion should have no connection with the excess rules and cannot remove them

Reply:
fixed
Test was rewritten